zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterEventPoller.java

```diff
@@ -72,6 +72,8 @@ public void setInterpreterGroup(InterpreterGroup interpreterGroup) {
   @Override
   public void run() {
     Client client = null;
+    AppendOutputRunner.setListener(listener);
```

Comment:
Having a static setter does not look good to me - it hides the state which is hard to mock and test.

Would it be possible to have an instance of `AppendOutputRunner` here instead please?